ecorest/endpoints/eco_full.go

```diff
@@ -0,0 +1,80 @@
+package endpoints
+
+import (
+	"fmt"
+	"github.com/OpenTreeMap/otm-ecoservice/eco"
+	"github.com/OpenTreeMap/otm-ecoservice/ecorest/cache"
+	"strconv"
+	"time"
+)
+
+type FullBenefitsPostData struct {
+	Region      string
+	Query       string
+	Instance_id string
+}
+
+// We can't marshall maps directly with
+// go-rest so we just wrap it here
+type FullBenefitsWrapper struct {
+	Benefits map[string]map[string]float64
+}
+
+func EcoFullBenefitsPOST(cache *cache.Cache) func(*FullBenefitsPostData) (*FullBenefitsWrapper, error) {
```

Comment:
This file is nearly identical to `eco_summary.go`. Why not have the outer function also take a benefit-calculator function to close over, or, if that makes the signatures inconvenient, introduce another layer of currying or a helper function.